### ISSUE TITLE
swcMinify causes some problems

### ISSUE THREAD

**Issue by lemonied** [vercel/next.js#40601]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro
Binaries:
  Node: 16.17.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant packages:
  next: 12.3.1-canary.2
  eslint-config-next: 12.3.0
  react: 18.2.0
  react-dom: 18.2.0

### What browser are you using? (if relevant)

Google Chrome 105.0.5195.102

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

In development mode everything works fine.
Program crashed when I execute `npm run build` and `npm run start`，However when I set `swcMinify ` to `false` everything works fine again

crash in production mode with `swcMinify: true`:
![image](https://user-images.githubusercontent.com/25240445/190556926-f70cdd82-2483-4eeb-b7e1-cd1e1481aa08.png)

### Expected Behavior

I expect the program to behave the same on `npm run dev` and `npm run start`

### Link to reproduction

https://codesandbox.io/s/agitated-tree-t3jml4?file=/pages/index.tsx

### To Reproduce

```tsx
import Highlight, { defaultProps } from "prism-react-renderer";

const exampleCode = `
export class Store<T> {
  public map<I>(i: I): I {
    return i;
  }
}

export const createStore = <T>(defaultState: T) => {
  return new Store(defaultState);
};
`;

export default function Home() {
  return (
    <Highlight {...defaultProps} code={exampleCode} language="typescript">
      {({ className, style, tokens, getLineProps, getTokenProps }) => (
        <pre className={className} style={style}>
          {tokens.map((line, i) => (
            <div key={i} {...getLineProps({ line, key: i })}>
              {line.map((token, key) => (
                <span key={key} {...getTokenProps({ token, key })} />
              ))}
            </div>
          ))}
        </pre>
      )}
    </Highlight>
  )
}
```
The above code works fine in the `development mode`，crash in production mode with `swcMinify: true` but works fine with `swcMinify: false`

**Comment by lemonied** [comment]
This problem occurs when I upgrade the version to `12.3.0`，everything works fine in `12.2.5`

**Comment by lemonied** [comment]
can someone help me :monocle_face:

**Comment by sushantdhiman** [comment]
I can also confirm similar issue with `swcMinify: true`

```js
Uncaught (in promise) TypeError: h is undefined
    NextJS 16
        nhw
        n_U
        nfq
        td
        td
        pf
        pf
        pf
        Ze
        Ze
        dispatch
        n
        dispatch
        dispatch
        saveDispatch
        postMessage
Caused by: undefined
[94d85b83-ad559083d4c6fd15.js:1:408967](https://thefamilytree.app/_next/static/chunks/94d85b83-ad559083d4c6fd15.js)
```

**Comment by lemonied** [comment]
> 

Are you also using version 12.3.0? I have this problem after upgrading to 12.3.0

**Comment by lemonied** [comment]
@sushantdhiman  Are you also using version 12.3.0? I have this problem after upgrading to 12.3.0

**Comment by Austaras** [comment]
Should be duplicate of swc-project/swc#5846

**Comment by zzfn** [comment]
me too

**Comment by chentsulin** [comment]
You can try the latest canary https://github.com/vercel/next.js/releases/tag/v12.3.1-canary.3 and see if the problem still there.

**Comment by zzfn** [comment]
> You can try the latest canary https://github.com/vercel/next.js/releases/tag/v12.3.1-canary.3 and see if the problem still there. The fix of [swc-project/swc#5846](https://github.com/swc-project/swc/issues/5846) should be included in v12.3.1-canary.3.

no,it still break when i use `import multiavatar from '@multiavatar/multiavatar/esm'`
```
{
    "@multiavatar/multiavatar": "^1.0.7",
}
```

**Comment by lemonied** [comment]
> You can try the latest canary https://github.com/vercel/next.js/releases/tag/v12.3.1-canary.3 and see if the problem still there. The fix of [swc-project/swc#5846](https://github.com/swc-project/swc/issues/5846) has been included in v12.3.1-canary.3.

It's useful for me, the problem disappeared in version v12.3.1-canary.3, Thanks!
